drivers/media/i2c/irs1125_regs.h

```diff
@@ -0,0 +1,4126 @@
+/*
+ * IRS1125x All-in-One firmware v.2126
+ *
+ * Copyright Infineon Technologies AG 2017-2019.
+ * Distributed under the Boost Software License, Version 1.0.
+ * (See accompanying file LICENSE_1_0.txt or copy at
+ * https://www.boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef _IRS1125_FIRMWARE_H_
+#define _IRS1125_FIRMWARE_H_
+
+#include <linux/types.h>
+
+struct regval_list {
+	u16 addr;
+	u16 data;
+};
+
+struct regval_list irs1125_alternate_fw[] = {
```

Comment:
Wowsers, 4kB of firmware as a list of register writes.

There are firmware loading mechanisms that may be better suited to doing this with a binary firmware file that typically lives in /lib/firmware. Your driver loads it via request_firmware(). See the majority of the DVB drivers for examples (one example at random https://github.com/raspberrypi/linux/blob/rpi-4.19.y/drivers/media/dvb-frontends/tda10071.c#L852).

Doing it this way also decouples the firmware from the driver. Admittedly that may be a bad thing should there be a hard dependency between the two.

If this does need to remain here then it definitely needs to gain a ```const``` to allow the compiler to optimise it.

Reply:
Cool, I did not know that. It's replaced now by require_firmware